target-chains/near/receiver/src/lib.rs

```diff
@@ -162,11 +166,12 @@ impl Pyth {
         Ok(())
     }
 
-    /// Return the deposit required to update a price feed.
-    pub fn get_update_fee_estimate(&self) -> u64 {
+    /// Return the deposit required to update a price feed. This is the upper limit for an update
+    /// call and any remaining deposit not consumed for storage will be refunded.
+    pub fn get_update_fee_estimate(&self) -> U128 {
         let byte_cost = env::storage_byte_cost();
```

Comment:
In the other contracts the fee varies depending on how many price feeds are being updated in a single batch, and the fee estimate functions take a list of the VAAs that will be updated.

(See https://github.com/pyth-network/pyth-crosschain/blob/main/target-chains/aptos/contracts/sources/pyth.move#L375-L380).

Reply:
Batch VAA's are difficult on NEAR because of the async nature, instead I thought better to allow the caller to simply chain their own promises of this method. Otherwise it needs to thread context through all the VAA verification and updates and calculate how to resolve deposit refunds at the end. I'm not sure It's worth it.

Reply:
that's right but still, we might want to have charge a variable rate fee per vaa. What Tom suggests is that this method should take the update data as input (although we don't use it here)